### PR TITLE
Fix check for unsupported columns if a column starts with a N/A value

### LINKED ISSUE #3073: OmniSci engine incorrectly treats N/A first element in a column as a marker of unsupported column
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **Modin version** (`modin.__version__`): 0.9.1
- **Python version**: 3.7
- **Code we can use to reproduce**:

```python
import pandas
from modin.experimental.engines.omnisci_on_ray.frame.partition_manager import OmnisciOnRayFrameManager
print('none-first', OmnisciOnRayFrameManager._get_unsupported_cols(pandas.DataFrame({'col': [None, 'a', 'b']})))
print('value-first', OmnisciOnRayFrameManager._get_unsupported_cols(pandas.DataFrame({'col': ['a', 'a', 'b']})))
```

### Describe the problem

Current output:
```
none-first (None, ['col'])
value-first (pyarrow.Table
col: string, [])
```

Expected output:
```
none-first (pyarrow.Table
col: string, [])
value-first (pyarrow.Table
col: string, [])
```

As can be seen here:

https://github.com/modin-project/modin/blob/6bc748e6e835ecfb6e9779aa93cbd47c2f412511/modin/experimental/engines/omnisci_on_ray/frame/partition_manager.py#L105-L113

`OmnisciOnRayFrameManager` only ever checks first element in each `object` column, and treats the whole column as "unsupported" if that value is not a string. However, N/A values (at least `None`) are perfectly fine, as the second `print()` shows.

This mis-treatment can cause other routines to unnecessary "default-to-pandas" reducing the performance.

### PR DESCRIPTION
**Pull request by vnlitvinov**
## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3073
- [x] tests added and passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py [hunk 1]
```diff
@@ -1709,5 +1713,24 @@ def applier(df, *args, **kwargs):
         run_and_compare(applier, data=self.data)
 
 
+class TestUnsupportedColumns:
```
Comment (dchigarev) [inline modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py:1716]: we already have a class, that stores tests for detecting unsupported columns  ([`TestBadData`](https://github.com/modin-project/modin/blob/6bc748e6e835ecfb6e9779aa93cbd47c2f412511/modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py#L1616)), do we need a new one?

**Comment by dchigarev** [comment]
LGTM, just one minor question

File: modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py (hunk 1 above)
Comment (vnlitvinov) [reply modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py:1716]: that class serves a different purpose from what I see, it checks how bad data is handled. I'm handling here "good" data, or rather I'm checking the _validation_ of is data good or bad...

Comment (dchigarev) [reply modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py:1716]: Makes sense

**Review by dchigarev — APPROVED** [review]

**Review by ienkovich — APPROVED** [review]
LGTM!

**Pull request merged**